google/provider.go

```diff
@@ -297,3 +298,39 @@ func convertStringArr(ifaceArr []interface{}) []string {
 	}
 	return arr
 }
+
+// flattenComputeMetadata transforms a list of MetadataItems (as returned via the GCP client) into a simple map from key
+// to value.
+func flattenComputeMetadata(metadata []*compute.MetadataItems) map[string]string {
+	m := map[string]string{}
+
+	for _, val := range metadata {
+		// check for duplicates
+		if val.Value == nil {
+			continue
+		}
+		if v2, ok := m[val.Key]; ok {
+			// warn loudly!
+			log.Printf("[WARN] Key '%s' already has value '%s' when flattening - ignoring incoming value '%s'",
+				val.Key,
+				v2,
+				*val.Value)
+		}
+		m[val.Key] = *val.Value
+	}
+
+	return m
+}
+
+// expandComputeMetadata transforms a map representing computing metadata into a list of compute.MetadataItems suitable
+// for the GCP client.
+func expandComputeMetadata(m map[string]string) []*compute.MetadataItems {
+	metadata := make([]*compute.MetadataItems, len(m))
+
+	idx := 0
+	for key, value := range m {
+		metadata[idx] = &compute.MetadataItems{Key: key, Value: &value}
```

Comment:
`idx++`? (you can also do `metadata = append(metadata, &compute.MetadataItems{Key: key, Value: &value})`. I think since the backing array won't need to be resized, it should have the same performance but I'm not 100% sure.)